src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzModule.groovy

```diff
@@ -229,31 +280,25 @@ class SnpLzModule extends AbstractHighDimensionDataTypeModule {
             fetchSize SNP_PATIENTS_FETCH_SIZE
             fetchMode 'patient', FetchMode.JOIN
         }
-        log.debug "Found ${sssList.size()} subject ordering entries for assays $assays"
+        log.debug "Found ${sssList.size()} subject ordering entries for assays."
 
         LinkedHashMap<AssayColumn, Integer> assayIndexMap = [:] as LinkedHashMap
-        Set<AssayColumn> foundAssays = [] as Set
-
-        def assayForPatientId = assays.collectEntries { [it.patient.id, it] }
-        // map assays to their positions in the blobs
-        sssList.each { SnpSubjectSortedDef sss ->
-            AssayColumn assay = assayForPatientId[sss.patient.id]
-            if (!assay) {
-                log.trace "SnpSubjectSortedDef entry $sss not matched to any selected assay"
-                return
+        Set<AssayColumn> assaysNotFound = [] as Set
+
+        /*
+         * Map assays to their positions in the blobs.
+         * The ordering of <var>assays</var> is preserved in the iterator of assayIndexMap.
+         */
+        def sssForPatientId = sssList.collectEntries { [it.patient.id, it] }
+        assays.each { AssayColumn assay ->
+            SnpSubjectSortedDef sss = sssForPatientId[assay.patient.id]
+            if (!sss) {
+                assaysNotFound << assay
```

Comment:
@gijskant I assume this should be assaysNotFound << assay rather than assaysNotFound << sss? Fixed it